### ISSUE TITLE
Problems when trying generate data

### ISSUE THREAD

**Issue by Hu-Yuch** [columbia-ai-robotics/scalingup#21]
Your program get stuck in this step when run "python scalingup/inference.py evaluation.num_episodes=50000 policy=scalingup evaluation=drawer",how can I deal with it
![96dbacbb544e85b5342fc7b1f963f18](https://github.com/real-stanford/scalingup/assets/102276943/7091160f-caa9-4160-bc5c-080fb9a17cc5)
?

**Comment by clorislili** [comment]
True. The speed of data generation process is quite slow. Any ways to enhance its speed? Thanks

**Comment by Louis-ZhangLe** [comment]
> Your program get stuck in this step when run "python scalingup/inference.py evaluation.num_episodes=50000 policy=scalingup evaluation=drawer",how can I deal with it ![96dbacbb544e85b5342fc7b1f963f18](https://private-user-images.githubusercontent.com/102276943/294663336-7091160f-caa9-4160-bc5c-080fb9a17cc5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYwMTY3MzcsIm5iZiI6MTcwNjAxNjQzNywicGF0aCI6Ii8xMDIyNzY5NDMvMjk0NjYzMzM2LTcwOTExNjBmLWNhYTktNDE2MC1iYzVjLTA4MGZiOWExN2NjNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyM1QxMzI3MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xODFjYjM1MWMwYjE5ZjgzZGNjNTZhY2FkZGVhMGVkNzc4NjI0ZTg2OTUyMWVhYjhmMTFkNWM4ZDZhYWM5NWNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.gYi4UIeVtxnWbPMiB45Q6LrEgguv7bM6WUsgmURP6GQ) ?

do you use the CACHE?

**Comment by huy-ha** [comment]
Hey @h129344010 and @clorislili ,

As @Louis-ZhangLe mentioned, if it's getting stuck without running, it's probably trying to make lots of OpenAI API calls before running the simulation. You can follow instructions [here](https://github.com/real-stanford/scalingup/blob/master/docs/walkthrough.md#-cache) if you don't have the cache setup. The latest version of the codebase is not working with the cache (#22 ) but the fix is to just checkout the earliest version of this codebase.

If you are already using cache, then rendering could help speed up simulation significantly. This is because rendering is used to generate pointclouds, which is used by the grasp and placement samplers.

From your logger output @h129344010 , it looks like you're ssh-ing into the node you're using to run data generation with X11 forwarding. I might be wrong, but this could slow down rendering significantly. Further, glfw doesn't work well with X11 forwarding (it failed to open the display in the log, so I suspect your pointclouds are actually coming out blank).

In our headless servers, we installed VirtualGL, which allows us to run headless rendering on the GPU. This is my recommendation. If you don't want to go through the hassle of installing VirtualGL, you could rely on a different OpenGL backend other than glfw. For instance, running the data generation with `MUJOCO_GL=osmesa` or `MUJOCO_GL=egl` in the environment variables should work well with CPU only rendering.

Hope this helps!
